model/riscv_insts_vext_utils.sail

```diff
@@ -448,13 +451,17 @@ function get_fixed_rounding_incr(vec_elem, shift_amount) = {
   if shift_amount == 0 then 0b0
   else {
     let rounding_mode = vcsr[vxrm];
+    assert(shift_amount <= 'm);
```

Comment:
Definitely add a `// TODO: Change the type signature to enforce this assertion.`

We shouldn't try to do it here though - it will likely make the PR much bigger.